README.md

```diff
@@ -1,92 +1,70 @@
+   * It will automatically be used as the HTTPS cert during local execution, thanks to `launchSettings.json`
+2. Create a CosmosDB database instance in Azure, using either the above instructions, or [manually](https://docs.microsoft.com/azure/cosmos-db/sql/create-cosmosdb-resources-portal).
+   * Navigate to the Keys section of your CosmosDB instance in the Azure portal to find your connection information.
+   * If you've used the ARM templates as described above, your Database will be named `WinGet` and your Collection will be `Manifests`
+3. Copy `src\WinGet.RestSource.Functions\local.settings.template.json` to `local.settings.json` and populate required fields from the above Keys section.
+4. In source codes, change HttpTrigger level to Anonymous for InformationGet, ManifestSearchPost and ManifestGet endpoints.
+5. Run the `WinGet.RestSource.Functions` project locally in Visual Studio using F5.
+6. Add it as a source in winget with: `winget source add -n "winget-pkgs-restsource" -a https://localhost:7071/api/ -t "Microsoft.Rest"`
+
+Your commands to winget will now use your locally running REST instance as user added source.
+
+## Running Tests
+
+Running tests are a great way to ensure that functionality is preserved across major changes. You can run these tests in Visual Studio Test Explorer. In Visual Studio, run the tests from the menu with Test > Run All Tests
+
+### Unit Testing Prerequisites
+
+* Install the [Cosmos DB Emulator](https://docs.microsoft.com/azure/cosmos-db/local-emulator?tabs=ssl-netstd21)
+* Copy the `WinGet.RestSource.UnitTest\Test.runsettings.template.json` template configuration to `Test.runsettings.json`
+  * The defaults should work for your local Cosmos DB emulator instance. You can change the configuration to point to a Cosmos DB instance in Azure instead.
+  * Alternatively, all of the test configuration properties can be set as environment variables. This is useful for overriding properties in an ADO build.
+
+### Integration Testing Prerequisites
+
+* Install the [winget client](https://github.com/microsoft/winget-cli) locally.
+* Copy the `WinGet.RestSource.IntegrationTest\Test.runsettings.template.json` template configuration to `Test.runsettings.json`
+  * Modify the `RestSourceUrl` property to point to a deployed rest source endpoint. You can use the below instructions to deploy a rest instance.
+  * If the local winget client doesn't already have the source added, the integration tests can add it. To do so, change the `AddRestSource` property to true. Visual Studio must be running as admin in this case.
+  * There is a test case that **modifies** the rest source. By default it's disabled, to run it the `RunWriteTests` setting must be set to true. The `FunctionsHostKey` setting must also be set since the add/update/delete endpoints all require function authorization. We recommend creating a new pipeline-specific host key for this purpose.

```

Comment:
```suggestion
  * There is a test case that **modifies** the rest source. This test is disabled by default. To run it, the `RunWriteTests` setting must be set to true. The `FunctionsHostKey` setting must also be set since the add/update/delete endpoints all require function authorization. We recommend creating a new pipeline-specific host key for this purpose.

```